flow/aggregates.py

```diff
@@ -0,0 +1,475 @@
+        # aggregator object i.e. aggregator.groupsof(1) is created using the class method,
+        # then we explicitly set the `_is_aggregate` attribute to False.
+        self._is_aggregate = True
+        self._aggregator_function = aggregator_function
+        self._sort_by = sort_by
+        self._reverse_order = bool(reverse_order)
+        self._select = select
+        # The idea of providing users with an additional tag attribute comes from the fact
+        # that the byte code of an aggregator function cannot truly differentiate it with
+        # some other aggregator function. Hence while checking the equality, we also
+        # check the equality based on tag provided by an user or us in a classmethod.
+        self._tag = tag
+
+    @classmethod
+    def groupsof(cls, num=1, sort_by=None, reverse_order=False, select=None):
+        """Aggregates jobs of a set group size.
+
+        By default aggregate of a single job is created.
+
+        If the number of jobs present in the project is not divisible by the number
+        provided by the user, the last aggregate will be smaller and contain all
+        remaining jobs.
+        For instance, if 10 jobs are present in a project and they are aggregated in
+        groups of 3, then the generated aggregates will have lengths 3, 3, 3, and 1.
+
+        The code block below provides an example on how jobs can be aggregated in
+        groups of 2.
+
+        .. code-block:: python
+
+            @aggregator.groupsof(num=2)
+            @FlowProject.operation
+            def foo(*jobs):
+                print(len(jobs))
+
+        :param num:
+            The default size of aggregates excluding the final aggregate.
+        :type num:
+            int
+        :param sort_by:
+            Before aggregating, sort the jobs by a given statepoint parameter.
+            The default behavior is no sorting.
+        :type sort_by:
+            str or NoneType
+        :param reverse_order:
+            States if the jobs are to be sorted in reverse order.
+            The default value is False.
+        :type reverse_order:
+            bool
+        :param select:
+            Condition for filtering individual jobs. This is passed as the
+            callable argument to `filter`.
+            The default behavior is no filtering.
+        :type select:
+            callable or NoneType
+        """
+        try:
+            if num != int(num):
+                raise ValueError('The num parameter should be an integer')
+            num = int(num)
+            if num <= 0:
+                raise ValueError('The num parameter should have a value greater than 0')
+        except TypeError:
+            raise TypeError('The num parameter should be an integer')
+
+        # The idea of this method was originally implemented in #52 which was copied from:
+        # https://docs.python.org/3/library/itertools.html#itertools.zip_longest
```

Comment:
Putting #52 doesn't help very much, it requires somebody to know that it refers to a pull request (and in the event that we ever switched off of Github we could lose PRs entirely). I would include the link and mention the `grouper` recipe (that's the function at that link that implements this technique).